src/core/qgsproject.h

```diff
@@ -951,6 +975,8 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
 
     QMap<QString, QgsMapLayer *> mMapLayers;
 
+    QgsWeakMapLayerPointerList mLayerOrder;
```

Comment:
While we are trying to move away from QgsProject as singleton - and that's a good move! - it will still keep being something similar, as in there is always one QgsProject per workspace/project. So everything that is in QgsProject will be available only once per workspace.

As such it comes with very similar properties as a singleton: it's very easy to implement but doesn't allow for flexibility. Therefore I'd like to reduce the number of direct members of QgsProject to a minimum and am looking for some other place to put this.

Apart from this (in practice we also only have one QgsMapThemeCollection per QgsProject) I would like to avoid the QgsProject to become a "general dumping ground" for everything.